llvm/lib/Target/STM8/STM8InstrFormats.td

```diff
@@ -20,14 +20,89 @@ class STM8Inst<dag outs, dag ins, string asmstr, list<dag> pattern, int size>
   let AsmString = asmstr;
   let Pattern = pattern;
 
-  int Size = size;
+  int ArgSize = -1;  // argument size in bytes
+  bits<8> Op = op;
+  bits<8> Mod = mod;
+  bits<32> Args = 0;
+
+  // even though we have a CISC ISA here, the longest instruction will
+  // always only be 6 bytes long. As there also exists a clear opcode
+  // structure it is better to postprocess the instruction to bring it
+  // to the right size, instead of rewriting the same few lines everywhere.
```

Comment:
CISC and RISC are vaguely-defined terms, each having multiple characteristics, and fixed-length instruction is only one of them (and a minor one, I think). Another is that RISC architectures usually have a larger number of registers. The main criterion for a CISC architecture, as I understand it, is that the CPU is able to do many things in one instruction (usually implying that it is not a load-store architecture). An example for such an instruction is x86' `rep movsb`, which is basically memcpy in a single instruction. STM8 does not have such instructions.

Overall I'd say that STM8 doesn't fit cleanly in either of these categories and describing it as either RISC or CISC is not useful.

Reply:
Okay thanks. So just throw the CISC mentions out of there?